code/aepcomposeui/src/main/java/com/adobe/marketing/mobile/aepcomposeui/aepui/AEPUI.kt

```diff
@@ -0,0 +1,22 @@
+package com.adobe.marketing.mobile.aepcomposeui.aepui
+
+import androidx.compose.runtime.mutableStateOf
+import com.adobe.marketing.mobile.aepuitemplates.AEPUITemplate
+import com.adobe.marketing.mobile.aepuitemplates.SmallImageTemplate
+import com.adobe.marketing.mobile.aepcomposeui.aepui.state.AEPUIState
+import com.adobe.marketing.mobile.aepcomposeui.aepui.state.SmallImageUIState
+
+
+/**
+ * Represents a UI component that can be rendered in the AepUI Engine. Binds a template data with a tracking observer.
+ * This is a sealed interface that can be implemented by different UI components like [SmallImageAEPUI], [LargeImageAEPUI], etc.
+ * This allows restricting the type of template and observer that can be associated with a UI component, which can be later
+ * used in the app to render the UI component via UIComposeExtensions.
```

Comment:
This was added in the POC for our internal understanding only. It should not be a part of the Kdocs